Sources/NIOCore/ByteBuffer-aux.swift

```diff
@@ -88,6 +88,30 @@ extension ByteBuffer {
         )
     }
 
+    // MARK: Hex encoded string APIs
+    /// Write `string` into this `ByteBuffer` in ASCII hexadecimal, moving the writer index forward appropriately.
+    /// - parameters:
+    ///     - plainHexEncodedBytes: The hex encoded string to write.
+    /// - returns: The number of bytes written.
+    @discardableResult
+    @inlinable
+    public mutating func writePlainHexEncodedBytes(_ plainHexEncodedBytes: String) throws -> Int {
+        var slice = plainHexEncodedBytes.utf8[...]
+        var written = 0
+        var bytesToWrite: [UInt8] = []
+
+        // Single pass to ensure string is a valid representation of hex bytes
+        while let nextByte = try slice.popNextHexByte() {
+            bytesToWrite.append(nextByte)
+        }
+
+        for byte in bytesToWrite {
+            written += self.writeInteger(byte)
+        }
```

Comment:
We don't need the intermediate array. Instead of using `bytesToWrite` and calling `append`, just call `writeInteger` directly.

Reply:
XCTAssertThrowsError(try buffer.writePlainHexEncodedBytes("68 1"))

If I write it the way you have suggested, this method will throw, BUT, the 68 will be written to the byte. Would that not leave the byte in an inconsistent or unintended state potentially leading to some confusion?

Reply:
We can save the original writer index and restore it when we throw the error.

Reply:
Thanks, will do